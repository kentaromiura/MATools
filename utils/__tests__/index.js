const Transition = require('@kentaromiura/transition');
const cubicBezier = require('..').cubicBezier;

test('should work', (done) => {
    const trans = new Transition(
        4000, 
        cubicBezier(
            [
                { x: 0, y: 0}, { x: 0, y: 1.3}, { x: 0.48, y: 1.36}, { x: 0.5, y: 0}, { x: 0.5, y: 1}, { x: 1, y: 1}, { x: 1, y: 0}
            ], 0.0001),
        (t) => {
            expect(t).toMatchSnapshot();
        }
    );
    
    trans.onComplete(() => {
        expect(true).toBe(true);
        done()
    });

    trans.step(0, true);
    trans.step(1000, true);
    trans.step(2000, true);
    trans.step(3000, true);
    trans.step(4000, true);
        
})
