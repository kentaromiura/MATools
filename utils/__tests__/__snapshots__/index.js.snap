// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work 1`] = `0`;

exports[`should work 2`] = `0.9975439490436884`;

exports[`should work 3`] = `0.003980312515050173`;

exports[`should work 4`] = `0.7499999552965164`;

exports[`should work 5`] = `0`;
